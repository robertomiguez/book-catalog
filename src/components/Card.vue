<template>
    <div>
        <div class=title>{{ title }}</div>
        <img
          :src="img"
          alt=""
          v-if="img"
          width=150
          height=200
        />
        <div class=description>{{ description }}</div>
    </div>
</template>

<script>
export default {
  name: `Card`,
  props: {
    title: {
      type: String,
      required: true
    },
    img: {
      type: String,
      required: true
    },
    description: {
      type: String,
      required: true
    }
  }  
}
</script>

<style>
.title {
    font-style: italic;
    font-family: fantasy;
    font-weight: 1000;
    height: 45px;
}
.description {
    height: 130px;
    padding-top: 10px;
}

</style>