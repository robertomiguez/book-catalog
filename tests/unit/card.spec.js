import { shallowMount } from '@vue/test-utils'
import Card from '@/components/Card.vue'

let wrapper

beforeEach(() => {
  wrapper = shallowMount(Card, {
    propsData: { 
      title: `card title`, 
      img: `img.jpg`,
      description: `blah` 
    },
    mocks: {},
    stubs: {},
    methods: {}
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe(`Card`, () => {
  test(`testing component props`, () => {
    expect(wrapper.find(`.title`).text())
      .toBe(`card title`)
    expect(wrapper.find(`img`).attributes().src)
      .toBe(`img.jpg`)
    expect(wrapper.find(`.description`).text())
      .toBe(`blah`)
  })

  test(`regression test`, () => {
    expect(wrapper.element)
      .toMatchSnapshot()
  })  

})