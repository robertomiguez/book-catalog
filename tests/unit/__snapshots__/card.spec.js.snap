// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card regression test 1`] = `
<div>
  <div
    class="title"
  >
    card title
  </div>
   
  <img
    alt=""
    height="200"
    src="img.jpg"
    width="150"
  />
   
  <div
    class="description"
  >
    blah
  </div>
</div>
`;
